memor/session.py

```diff
@@ -288,11 +289,12 @@ def update_messages(self,
         :param status: status
         """
         _validate_list_of(messages, "messages", (Prompt, Response), "`Prompt` or `Response`")
-        self._messages = messages
         if status:
-            self.update_messages_status(status)
+            _validate_status(status, messages)
         else:
-            self.update_messages_status(len(messages) * [True])
+            status = len(messages) * [True]
+        self._messages = messages
+        self._messages_status = status
```

Comment:
Use `self.update_messages_status(status)` intead of `self._messages_status = status`